### ISSUE TITLE
jinja[invalid] does not recognize item in loops

### ISSUE THREAD

**Issue by rascasoft** [ansible/ansible-lint#3082]
##### Summary

When using loops ansible-lint is unable to evaluate item coming from loop, ending in a `jinja[invalid]` error.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
> ansible-lint --version
ansible-lint 6.13.1 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Define a yaml like this:

```yaml
- name: Test play
  hosts: localhost
  tasks:
    - name: Use with_items to populate a file
      ansible.builtin.copy:
        content: |
          {{ item.cfg | to_nice_yaml(indent=2, width=9999) }}
        dest: "/tmp/{{ item.name }}.yaml"
        mode: '0640'
      with_items:
        - name: file1
          cfg:
            variable_a: 'A'
            variable_b: 'B'
            mylist:
              - a
              - b
              - c
        - name: file2
          cfg:
            variable_a: 'C'
            variable_b: 'D'
            mylist:
              - d
              - e
              - f
```

Which gets executed with success:

```console
> ansible-playbook test.yaml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Test play] ******************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************
ok: [localhost]

TASK [Use with_items to populate a file] ******************************************************************************************************************
changed: [localhost] => (item={'name': 'file1', 'cfg': {'variable_a': 'A', 'variable_b': 'B', 'mylist': ['a', 'b', 'c']}})
changed: [localhost] => (item={'name': 'file2', 'cfg': {'variable_a': 'C', 'variable_b': 'D', 'mylist': ['d', 'e', 'f']}})

PLAY RECAP ************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

With expected results:

```console
> cat /tmp/file{1,2}.yaml
mylist:
- a
- b
- c
variable_a: A
variable_b: B
mylist:
- d
- e
- f
variable_a: C
variable_b: D
```

##### Desired Behavior

I see that yaml/jinja2 as legit.

##### Actual Behavior

Problem is that `ansible-lint` is not able to interpret the `item` elements:

```console
> ansible-lint test.yaml
INFO     Set ANSIBLE_LIBRARY=/home/rasca/.cache/ansible-compat/1c2905/modules:/home/rasca/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/rasca/.cache/ansible-compat/1c2905/collections:/home/rasca/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/rasca/.cache/ansible-compat/1c2905/roles:/home/rasca/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Using /home/rasca/.cache/ansible-compat/1c2905/roles/mmul.terraform_tfs_generator symlink to current repository in order to enable Ansible to find the role using its expected full name.
INFO     Executing syntax check on playbook test.yaml (0.43s)
WARNING  Listing 1 violation(s) that are fatal
jinja[invalid]: to_nice_yaml - 'item' is undefined. 'item' is undefined
test.yaml:6 Task/Handler: Use with_items to populate a file

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary               
 count tag            profile rule associated tags 
     1 jinja[invalid] basic   formatting           

Failed after min profile: 1 failure(s), 0 warning(s) on 1 files.
```

It says item is undefined, which is not.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
